src/phpDocumentor/Compiler/Linker/Linker.php

```diff
@@ -157,22 +167,24 @@ public function substitute($item, $container = null)
                 return null;
             }
 
-            $newContainer = ($this->isDescriptorContainer($item)) ? $item : $container;
+            $newContainer = $this->isDescriptorContainer($item) ? $item : $container;
 
             $this->processedObjects[$hash] = $hash;
 
             $objectClassName = get_class($item);
-            $fieldNames = $this->substitutions[$objectClassName] ?? [];
+            $fieldNames      = $this->substitutions[$objectClassName] ?? [];
 
             foreach ($fieldNames as $fieldName) {
                 $fieldValue = $this->findFieldValue($item, $fieldName);
-                $response = $this->substitute($fieldValue, $newContainer);
+                $response   = $this->substitute($fieldValue, $newContainer);
 
                 // if the returned response is not an object it must be grafted on the calling object
-                if ($response !== null) {
-                    $setter = 'set' . ucfirst($fieldName);
-                    $item->{$setter}($response);
+                if ($response === null) {
+                    continue;
                 }
+
+                $setter = 'set' . ucfirst($fieldName);
```

Comment:
This hurts, needs to be changed at some point. But I need to think about how, please add a note that we fix this. in the future.